core/templates/pages/exploration-player-page/services/conversation-skin.service.ts

```diff
@@ -0,0 +1,155 @@
+// distributed under the License is distributed on an "AS-IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+/**
+ * @fileoverview Common service for the exploration conversation skin,
+ * extracting functionalities from the old skin to utilize in the new skin
+ * and minimize duplication of code.
+ */
+
+import {StateCard} from 'domain/state_card/state-card.model';
+import {Injectable} from '@angular/core';
+import {downgradeInjectable} from '@angular/upgrade/static';
+import {ContentTranslationLanguageService} from '../services/content-translation-language.service';
+import {ContentTranslationManagerService} from '../services/content-translation-manager.service';
+import {ExplorationPlayerStateService} from '../services/exploration-player-state.service';
+import {PlayerPositionService} from '../services/player-position.service';
+import {PlayerTranscriptService} from '../services/player-transcript.service';
+import {ExplorationPlayerConstants} from '../exploration-player-page.constants';
+import {WindowDimensionsService} from 'services/contextual/window-dimensions.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class ConversationSkinService {
+  // The minimum width, in pixels, needed to be able to show two cards
+  // side-by-side.
+  TIME_PADDING_MSEC = 250;
+  TIME_FADEIN_MSEC = 100;
+  TIME_NUM_CARDS_CHANGE_MSEC = 500;
+
+  isAnimatingToTwoCards!: boolean;
+  isAnimatingToOneCard!: boolean;
+
+  constructor(
+    private contentTranslationLanguageService: ContentTranslationLanguageService,
+    private contentTranslationManagerService: ContentTranslationManagerService,
+    private explorationPlayerStateService: ExplorationPlayerStateService,
+    private playerPositionService: PlayerPositionService,
+    private playerTranscriptService: PlayerTranscriptService,
+    private windowDimensionsService: WindowDimensionsService
+  ) {}
+
+  handleNewCardAddition(newCard: StateCard): void {
+    this.playerTranscriptService.addNewCard(newCard);
+    const explorationLanguageCode =
+      this.explorationPlayerStateService.getLanguageCode();
+    const selectedLanguageCode =
+      this.contentTranslationLanguageService.getCurrentContentLanguageCode();
+    if (explorationLanguageCode !== selectedLanguageCode) {
+      this.contentTranslationManagerService.displayTranslations(
+        selectedLanguageCode
+      );
+    }
+
+    let totalNumCards = this.playerTranscriptService.getNumCards();
+
+    let previousSupplementalCardIsNonempty =
+      totalNumCards > 1 &&
+      this.isSupplementalCardNonempty(
+        this.playerTranscriptService.getCard(totalNumCards - 2)
+      );
+
+    let nextSupplementalCardIsNonempty = this.isSupplementalCardNonempty(
+      this.playerTranscriptService.getLastCard()
+    );
+
+    if (
+      totalNumCards > 1 &&
```

Comment:
Maybe add a comment or extract into a variable with a meaningful name - what these 4 conditions combined mean? Same for the else-if below

Reply:
Done